ampersand-multifield-view.js

```diff
@@ -0,0 +1,153 @@
+'use strict';
+
+var find = require('lodash.find');
+var View = require('ampersand-view');
+
+var MultiFieldView = View.extend({
+  template: [
+    '<div>',
+      '<p class="label" data-hook="label"></p>',
+      '<div data-hook="multifields"></div>',
+    '</div>'
+  ].join(''),
+
+  bindings: {
+    'label' : [
+      {
+        hook: 'label'
+      },
+      {
+        type: 'toggle',
+        hook: 'label'
+      }
+    ]
+  },
+
+  props: {
+    fields: {
+      type: 'array',
+      required: true,
+      allowNull: false,
+    },
+
+    label: ['string', true, ''],
+    name: 'string'
+  },
+
+  initialize: function(spec) {
+    if (!spec || !spec.name) {
+      throw new Error('must pass in a name');
+    }
+
+    this.value = spec.value || {};
+    this.name = spec.name;
+    this.valid = true;
+    this.validCallback = spec.validCallback || function() {};
+
+    if (!this.fields) {
+      throw new Error('must have an array of fields');
+    }
+
+    this.fields.forEach(function(field) {
+      field.parent = this;
+    }, this);
+  },
+
+  render: function() {
+    this.renderWithTemplate(this);
+    this.fieldContainerEl = this.queryByHook('multifields');
+
+    this.fields.forEach(function(field) {
+      this.renderField(field, true);
+    }, this);
+
+    this.setValue(this.value);
+    this.checkValid(true);
+
+    return this;
+  },
+
+  renderField: function(fieldView, renderInProgress) {
+    if (fieldView.rendered || !this.fieldContainerEl) {
+      return this;
+    }
+
+    if (!this.rendered && !renderInProgress) {
+      return this;
+    }
+
+    fieldView.render();
+    this.fieldContainerEl.appendChild(fieldView.el);
+  },
+
+  setValid: function(now, forceFire) {
+    var prev = this.valid;
+
+    this.valid = now;
+
+    if(prev !== now || forceFire) {
+      this.validCallback(now);
+    }
+  },
+
+  checkValid: function(forceFire) {
+    var valid = this.fields.every(function(field) {
+      return field.valid;
+    });
+
+    this.setValid(valid, forceFire);
+    return valid;
+  },
+
+  setValue: function(value) {
+    var fields = this.fields;
+    var findField = function(name) {
+      var field = find(fields, {name: name});
+
+      if (!field) {
+        throw new Error('Field ' + name + ' not found');
+      }
+
+      return field;
+    };
+
+    Object.keys(value).forEach(function(k) {
+      var field = findField(k);
+      field.setValue(value[k], !field.required);
+    });
+  },
+
+  update: function(field) {
+    this.trigger('change:' + field.name, field);
+
+    if (field.valid) {
+      this.checkValid();
```

Comment:
Maybe a better name is `updateValidity()`. I don't think the intent of the function is to answer a question, but rather determine if all fields are valid, set the `valid` property, and if the property changed, call the callback. The name `checkValid` comes from Ampersand's form-view implementation, and I borrowed from that design heavily since the MultiFieldView needs to act as a form for its fields.

Breaking it up as you suggested would mean having to explicitly call `this.setValid` in the `update`, `render`, and `initialize` functions (I need to fix it for the initialize, I realized I was setting `valid` directly).

Is your worry primarily with the name? A better solution might be to have a third `updateValid` function that is called where `checkValid` is now. `checkValid` would be renamed to `isValid` and only return the boolean, and then `setValid` would remain the same. `updateValid` would call `isValid` and then pass the result into `setValid`. Thoughts?